Sources/TuistKit/Services/Inspect/GraphImportsLinter.swift

```diff
@@ -0,0 +1,86 @@
+import TuistCore
+import TuistLoader
+import XcodeGraph
+
+enum InspectType {
+    case redundant
+    case implicit
+}
+
+final class GraphImportsLinter {
+    private let targetScanner: TargetImportsScanning
+
+    init(targetScanner: TargetImportsScanning = TargetImportsScanner()) {
+        self.targetScanner = targetScanner
+    }
+
+    func lint(
+        graphTraverser: GraphTraverser,
+        inspectType: InspectType
+    ) async throws -> [LintingIssue] {
```

Comment:
I'd consider splitting the method and its implementation into multiple ones to make the code easier to follow.

I'd envision something like:
```swift
func lintImplicitImports(graphTraverser: GraphTraversing) {
  let targetsToLint = graphTraverser.allInternalTargets().union(graphTraverser.allExternalTargets()).filter { ... }
  return targetImportsMap.flatMap { (target, dependencyImports) in
   let explicitTargetDependencies = explicitTargetDepencies(target)
    dependencyImports.map { dependencyImport in
      if !explicitTargetDependencies.contains(where: { $0.name == dependencyImport.name }) {
        return LintingIssue(...)
     }
    }
  }
}

struct DependencyImport {
  let name: String
  let count: Int
}

/// - Returns: A map of targets with dependency imports
private func targetImportsMap(graphTraverser: GraphTraversing, targetsToLint: [Target]) -> [Target: [DependencyImport]] {
  ...
}
```